reconcile/quay_permissions.py

```diff
@@ -77,48 +78,54 @@ def run(dry_run):
-                            quay_api.get_repo_team_permissions(
-                                repo_name, team_name)
-                        if current_role != role:
-                            logging.info(
-                                ['update_role', org_key, repo_name,
-                                 team_name, role])
-                            if not dry_run:
-                                try:
+                    try:
+                        permissions = team['permissions']
+                        role = team['role']
+                        for permission in permissions:
+                            if permission['service'] != 'quay-membership':
+                                logging.warning('wrong service kind, '
+                                                'should be quay-membership')
+                                continue
+
+                            perm_org_key = (
+                                permission['quayOrg']['instance']['name'],
+                                permission['quayOrg']['name']
+                            )
+
+                            if perm_org_key != org_key:
+                                logging.warning(
+                                    f'wrong org, should be {org_key}'
+                                )
+                                continue
+
+                            team_name = permission['team']
+
+                            # processing team section
+                            logging.debug(['team', team_name])
+
+                            current_role = \
+                                quay_api.get_repo_team_permissions(
+                                    repo_name, team_name)
+                            if current_role != role:
+                                logging.info(
+                                    ['update_role', org_key, repo_name,
+                                     team_name, role])
+                                if not dry_run:
                                     quay_api.set_repo_team_permissions(
                                         repo_name, team_name, role)
-                                except Exception as e:
-                                    error = True
-                                    logging.error(
-                                        'could not set repo permissions: '
-                                        'repo name: %s, '
-                                        'team name: %s. '
-                                        'details: {%s}',
-                                        repo_name, team_name, e
-                                    )
+                    except HTTPError:
+                        error = True
+                        logging.exception(
+                            f'Could not manage repo permissions on {repo_name}'
+                            f' - team {team_name}'
+                        )
+                    except Exception:
```

Comment:
I see no different treatment in the catch-all or in the `HTTPError`. Why the difference? If we reduced the scope of the `try` block, could we narrow the exceptions here?

Reply:
It started with somewhat different messages, and then converged, sorry, I'll simplify this.